src/Test/Hspec/Hedgehog.hs

```diff
@@ -142,8 +142,8 @@ hedgehog = id
 -- of 'Example' for a function for more details.
 --
 -- @since 0.0.0.0
-instance Example (PropertyT IO ()) where
-    type Arg (PropertyT IO ()) = ()
+instance m ~ IO => Example (PropertyT m ()) where
```

Comment:
So the main issue here is that end users won't be able to define their own instances of `Example (PropertyT TheirNeatMonad ())`

Which I'm pretty cool with, I think? But worth noting that it's a breaking change.

Wondering if putting an inference constraint on the `()`, too:

```suggestion
instance (a ~ (), m ~ IO) => Example (PropertyT m a) where
```

This would forbid users from writing instances like:

```haskell
instance (m ~ IO) => Example (PropertyT m (SqlPersistT m (TestT m ()))) where
  type Arg _ = SqlBackend

  -- run the property, 
  -- run the action,
  -- run the asserts
```

Which - idk - maybe it's good and proper to make that less magical?

Reply:
> Wondering if putting an inference constraint on the `()`, too

I'm trying to mentally piece together what the tangible benefits would be. I guess you could put an `undefined` as the last statement of a do-block and things may still type check. You may also get better type errors at times.

From what I understand, adding this constrained does not change the picture fundamentally. But it could be nice to have.

For me, the essential thing is to constrain the base monad to IO. This is what I have been using for the last couple of months.  Getting this on Hackage will remove a lot of friction points for me.